src/cc/frontends/clang/b_frontend_action.cc

```diff
@@ -255,6 +261,12 @@ bool ProbeVisitor::VisitMemberExpr(MemberExpr *E) {
     error(base->getLocEnd(), "internal error: MemberLoc is invalid while preparing probe rewrite");
     return false;
   }
+
+  // Checks to see if the expression references something that needs to be run
+  // through bpf_probe_read.
+  if (!ProbeChecker(base, ptregs_).needs_probe())
```

Comment:
Moving this check after we retrieve `base` is necessary the base points to an external address and not the structure member. For example, we want to replace `lp->ext_ptr` with a call to `bpf_probe_read` only if `lp` points to an external address.